scripts/services/migrations/up-to-0_18_4_0.ts

```diff
@@ -0,0 +1,28 @@
+import { types as T, matches } from "../../deps.ts";
+const { dictionary, boolean, shape, string, any } = matches;
+
+export const migration_up_to_0_18_4_0 = (config: T.Config): T.Config => {
+  if (Object.keys(config).length === 0) {
+    // service was never configured
+    return config;
+  }
+
+  if (
+    matches
+      .shape({
+        advanced: matches.shape({
+          p2p: matches.shape({ spynodebanlist: matches.boolean }, [
+            "spynodebanlist",
+          ]),
```

Comment:
```suggestion
          p2p: matches.shape({ spynodebanlist: matches.boolean }),
```

you dont want this to be optional when checking if it exists